js/impress.js

```diff
@@ -379,11 +386,23 @@
 
         // `goto` API function that moves to step given as `el` parameter (by index, id or element).
         // `duration` optionally given as second parameter, is the transition duration in css.
-        var goto = function( el, duration ) {
+        // `reason` is the string "next", "prev" or "goto" (default) and will be made available to
+        // preStepLeave plugins.
+        // `origEvent` may contain event that caused the calll to goto, such as a key press event
+        var goto = function( el, duration, reason, origEvent ) {
```

Comment:
Right now it doesn't seem like `reason` or `origEvent` are needed - especially `reason` seems very arbitrary with 3 possible values.
But I assume they are necessary for some plugins to tell the difference of what triggered the transition?

Reply:
For example the goto plugin needs origEvent, because you can tie the pressed key to go to an arbitrary slide (e.g. press 5 to go to step-5). So it needs to have access to the event that generated the transition to know whether it was a keypress. stop plugin uses reason. It only stops next events, but doesn't prevent you going backwards to prev.